search/searcher/search_conjunction.go

```diff
@@ -34,26 +33,32 @@ func init() {
 }
 
 type ConjunctionSearcher struct {
-	indexReader index.IndexReader
-	searchers   OrderedSearcherList
-	queryNorm   float64
-	currs       []*search.DocumentMatch
-	maxIDIdx    int
-	scorer      *scorer.ConjunctionQueryScorer
-	initialized bool
-	options     search.SearcherOptions
-	bytesRead   uint64
+	indexReader     index.IndexReader
+	searchers       []search.Searcher
+	queryNorm       float64
+	queryNormForKNN float64
+	currs           []*search.DocumentMatch
+	maxIDIdx        int
+	scorer          *scorer.ConjunctionQueryScorer
+	initialized     bool
+	options         search.SearcherOptions
+	bytesRead       uint64
 }
 
 func NewConjunctionSearcher(ctx context.Context, indexReader index.IndexReader,
 	qsearchers []search.Searcher, options search.SearcherOptions) (
 	search.Searcher, error) {
 	// build the sorted downstream searchers
-	searchers := make(OrderedSearcherList, len(qsearchers))
+	sortedSearchers := &OrderedSearcherList{
+		searchers: make([]search.Searcher, len(qsearchers)),
+		index:     make([]int, len(qsearchers)),
```

Comment:
what is the purpose of this?

Reply:
i had put the conjunction stuff in a different branch thinking i will put it out once the operator code was merged in, otherwise it will be unused code. But yea i will just put it here now itself

Reply:
disj heap searcher does not sort the searchers and hence original positions are retained in matchingIdx - an existing variable so i didnt need to do this operation there